runtime-decompiler/src/main/java/org/jrd/frontend/MainFrame/BytecodeDecompilerView.java

```diff
@@ -143,6 +138,29 @@ protected Void doInBackground() throws Exception {
             }
         });
 
+        filteredClassesJlist.addKeyListener(new KeyAdapter() {
+            @Override
+            public void keyReleased(KeyEvent e) {
+                if (e.getKeyCode() == KeyEvent.VK_UP || e.getKeyCode() == KeyEvent.VK_DOWN) {
+                    final String name = filteredClassesJlist.getSelectedValue();
+                    if (name != null || filteredClassesJlist.getSelectedIndex() != -1) {
+                        new SwingWorker<Void, Void>() {
+                            @Override
+                            protected Void doInBackground() throws Exception {
+                                try {
```

Comment:
Can not be reused the original worker which is already done for mouse listener?
Also... what about pgup/pgdown? 
Also, how does it behave when you are already on top/bottom of the list? I guess the item reload anyway. (but this bug may be hidden already in original mouselistener)
How does it behaves on single class programs? Maybe click/space/enter can enforce refresh?

Reply:
I can extract the worker into it's own function and call it from both mouse listener and key listener.

Pgdown/pgup does not trigger update, will fix.

When on top/bottom, it does trigger an update, same as clicking the same class again with mouse. Does it need fixing?

On a single class program, clicking the class/pgup/pgdown/arrow up/down would refresh it. Can implement enter as a trigger for update.

Reply:
Please extract the worker.
pgu/dow - ty!
the top/bottom/single  behaviour sounds good.